### DISCUSSION TITLE
NGU work with Space Invaders

### DISCUSSION THREAD

**Discussion by AlanBlanchet** [opendilab/DI-engine#766] [Q&A] [answered]
Hello community,

I tried making the `SpaceInvaders-v4` env work with the ngu policy implementation. I didn't manage to get it working and got lots of problems during the setup :

I went in the examples directory and copied example files, changed the env and the policy. And when I launched it gives me errors.

Some errors originate from the EnvWrapperV2 in the `inference` function.  The treetensor str keys get changed to int keys if I understand correctly. But the _forward_collect method of NGU tries gettings str keys from the ready_obs attribute.

Here are all the things I tried to do to get this working : 
- Changed collectors
- Changed the original config
  => Oh yeah, theres also an assert statement in NGU that checks if the "unroll_len" key is present and raises an error if so. But if I don't add the key the collector complains that the key is missing
- Downgrading di-engine versions until 0.4.7 and testing

Could anyone show me how to setup an NGU with SpaceInvaders-v4 ?
Many thanks

**Comment by AlanBlanchet** [comment]
I also directly tried configuring the serial_pipelines for ngu in the entry directory. But it also didn't work

**Comment by puyuan1996** [comment, marked as answer]
Hello, thank you for your interest. Please note that the configs for NGU on the Atari environments can be found [here](https://github.com/opendilab/DI-engine/tree/main/dizoo/atari/config/serial). For the config on SpaceInvaders, please refer to this [PR](https://github.com/opendilab/DI-engine/pull/767). Feel free to discuss any issues here. Best wishes.

**Reply by AlanBlanchet** [reply]
Thank you for your quick answer !
I'll try make it work for me today 😄.
